.travis.yml

```diff
@@ -1,13 +1,35 @@
 language: node_js
 node_js:
-  - '7'
-  - '6'
-  - '4'
+  - 9
+  - 8
+  - 6
+  - 4
+env:
+  - FRESH_DEPS=false
+  - FRESH_DEPS=true
+matrix:
+  exclude:
+    - node_js: 9
+      env: FRESH_DEPS=true
+    - node_js: 6
+      env: FRESH_DEPS=true
+    - node_js: 4
+      env: FRESH_DEPS=true
 cache:
   directories:
     - $HOME/.npm
-    - node_modules
-before_install: 'npm install --global npm'
-before_script: 'npm prune'
-script: 'npm run coverage'
-after_script: 'cat ./coverage/lcov.info | ./node_modules/.bin/coveralls'
+before_install:
+  - npm install --global npm@5.6.0
+  - npm --version
+  - md5sum package-lock.json > package-lock.json.md5
+install: |
+  if [[ ${FRESH_DEPS} == "true" ]]; then
+    npm install --no-shrinkwrap --prefer-online;
+  else
+    npm install --prefer-offline;
+    if ! md5sum --quiet -c package-lock.json.md5; then
+      echo "package-lock.json was modified unexpectedly. Please rebuild it using npm@$(npm -v) and commit the changes.";
+      exit 1;
+    fi
+  fi
```

Comment:
This logic could probably be packaged up into an npm package. Hint hint ;)

Reply:
Heh yes, I know. I'm waiting for https://www.npmjs.com/package/cipm to mature and ship with npm itself. At that point one could write a package that can be added as a dev dependency, is installed through `npm ci` (which won't update the lock file), and then use the new `--package-lock-only` flag to get the correct lock file and compare.